src/app/system/configure-data-fields/data-fields/data-fields.component.ts

```diff
@@ -0,0 +1,211 @@
+  fieldConfigurationId: any;
+  dataSource: any;
+  updatedFieldsData: any = [];
+  datatableName: any;
+  fieldName: any;
+
+  columnType: any;
+  columnTypeList: any[] = [
+    { name: "String" },
+    { name: "Number" },
+    { name: "Decimal" },
+    { name: "Boolean" },
+    { name: "Date" },
+    { name: "DateTime" },
+    { name: "Text" },
+    { name: "Dropdown" },
+  ];
+  columnTypeListFiltered: any = [];
+
+  displayedColumns = ["field", "isEnabled", "isMandatory", "isUnique","actions"];
+  addDisplayedColumns = ["name", "type", "mandatory", "isUnique", "length","code","actions"];
+
+  addDataSource = new BehaviorSubject<AbstractControl[]>([]);
+  rows: FormArray = this.fb.array([]);
+  form: FormGroup = this.fb.group({ tableArray: this.rows });
+  @ViewChild(NgModel, { static: true }) addField: NgModel;
+
+  constructor(
+    private route: ActivatedRoute,
+    private systemService: SystemService,
+    private router: Router,
+    private fb: FormBuilder,
+    private alertService:AlertService
+  ) {
+    this.route.data.subscribe((data: { columnCodes: any }) => {
+      this.codes = data.columnCodes;
+    });
+
+    this.route.paramMap.subscribe((paramMap) => {
+      this.countryId = paramMap.get("countryId");
+      this.type = paramMap.get("id");
+    });
+  }
+
+  ngOnInit(): void {
+    this.columnTypeListFiltered = this.columnTypeList;
+    this.getFieldConfigurationByType();
+    this.getFieldBaseData();
+  }
+
+  getFieldConfigurationByType() {
+    this.systemService.getFieldConfigurationByType(this.type, this.countryId).subscribe((res: any) => {
+      this.subModuleFields = res;
+      let temp = res.filter((value, index, self) => index === self.findIndex((t) => t.subentity === value.subentity));
+      this.subModule = temp;
+      this.subModuleFiltered = temp;
+      this.fieldConfigurationId = temp[0];
+      this.selectedSubModule(this.fieldConfigurationId?.subentity);
+    });
+  }
+
+  getFieldBaseData() {
+    this.systemService.getFieldConfigurationBasedata().subscribe((res: any) => {
+      this.baseData = res;
+    });
+  }
+
+  public isFiltered(field: any) {
+    if (field && field.fieldConfigurationId)
+      return this.subModuleFiltered.find((item) => item.fieldConfigurationId === field.fieldConfigurationId);
+  }
+
+  public filterColumnType(field: any) {
+    if (field && field.name && this.columnTypeListFiltered) {
+      return this.columnTypeListFiltered.find((item) => item.name === field.name);
+    } else {
+      return this.columnTypeList.find((item) => item.name === field.name);
+    }
+  }
+
+  subEntity:any;
+  selectedSubModule(subEntity: any) {
+    this.subEntity=subEntity;
+    this.datatableName = "m_field_" + subEntity.toLowerCase() + "_" + this.type.charAt(0).toLowerCase();
+    if (subEntity && this.subModule.length > 0) {
+      this.updatedFieldsData = this.subModuleFields.filter((i) => i.subentity === subEntity);
+    }
+  }
+
+  toggleMatTable: boolean = false;
+  addControl() {
```

Comment:
Please check what's wrong here. Tried to add a new custom data fields and received the following error
![image](https://user-images.githubusercontent.com/97087814/228497374-af58ea97-04d9-4f87-8e36-dd03112bf71d.png)

Request:
Body: 
`{"apptableName":"m_ou","datatableName":"m_field_office_o","officeCountryId":"2","addColumns":[{"name":"test1","type":"String","mandatory":true,"isUnique":false,"length":"50","subEntity":"OFFICE","entity":"OU"}]}`
Details of the  error:
`{"developerMessage":"The request was invalid. This typically will happen due to validation errors which are provided.","httpStatusCode":"400","defaultUserMessage":"Validation errors exist.","userMessageGlobalisationCode":"validation.msg.validation.errors.exist","errors":[{"developerMessage":"The parameter apptableName must be one of [ m_loan, m_savings_account, m_client, m_group, m_center, m_office, m_savings_product, m_product_loan ] .","defaultUserMessage":"The parameter apptableName must be one of [ m_loan, m_savings_account, m_client, m_group, m_center, m_office, m_savings_product, m_product_loan ] .","userMessageGlobalisationCode":"validation.msg.datatable.apptableName.is.not.one.of.expected.enumerations","parameterName":"apptableName","value":null,"args":[{"value":"m_ou"},{"value":["m_loan","m_savings_account","m_client","m_group","m_center","m_office","m_savings_product","m_product_loan"]}]}]}`

Reply:
I checked both of these and seem to be working. The issue might be from the API end. I have faced such issues as well while testing.

Reply:
@SauravPriyadarshy The issue is on the payload...You are passing apptableName m_loantype instead of m_product_loan and m_ou when it should have been m_office.

Reply:
now I am getting the following error:
`{"developerMessage":"The request was invalid. This typically will happen due to validation errors which are provided.","httpStatusCode":"400","defaultUserMessage":"Validation errors exist.","userMessageGlobalisationCode":"validation.msg.validation.errors.exist","errors":[{"developerMessage":"The parameter columns is mandatory.","defaultUserMessage":"The parameter columns is mandatory.","userMessageGlobalisationCode":"validation.msg.FieldConfiguration.columns.cannot.be.blank","parameterName":"columns","value":null,"args":[{"value":null}]}]}`

Reply:
Hi @SauravPriyadarshy ,  you might need to consider reusing some of the existing function to assist on this issue

```
            scope.generateAppTableName = function () {

                let apptableName = "";

                if (scope.entityName === "OU") {
                    apptableName = "m_office";
                } else if (scope.entityName === "ADDRESS" || scope.entityName === "CLIENT") {
                    apptableName = "m_client";
                } else if (scope.entityName === "LOANTYPE") {
                    apptableName = "m_loan";
                }

                return apptableName;
            }
```

Reply:
@william000000 is this working now?

Reply:
Yes @amydoxym everything is tested and OK. kindly let me know if you may have noticed anything that needs a fix